route/route.go

```diff
@@ -191,6 +199,45 @@ func (r *Route) PathSelection() {
 	r.updateEqualPathCount()
 }
 
+// Euql compares Are two routes and return true if they are equal
+func (r *Route) Equal(other *Route) bool {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	a := r.pfx.Equal(other.pfx)
+	b := r.ecmpPaths == other.ecmpPaths
+	c := true
```

Comment:
What's the purpose of "c"?

Reply:
don't ask... Refactored this function now